src/librustc_codegen_llvm/back/link.rs

```diff
@@ -813,8 +804,8 @@ fn link_natively(sess: &Session,
                     .unwrap_or_else(|_| {
                         let mut x = "Non-UTF-8 output: ".to_string();
                         x.extend(s.iter()
-                                 .flat_map(|&b| ascii::escape_default(b))
-                                 .map(|b| char::from_u32(b as u32).unwrap()));
+                                  .flat_map(|&b| ascii::escape_default(b))
+                                  .map(|b| char::from_u32(b as u32).unwrap()));
```

Comment:
`b` is `u8`, so this can be

```suggestion
                                  .map(|b| char::from(b)));
```